Rejigger ClickAppNotifications page

UniversalSuperBox opened the pull request:
Turns out there's a switch in ClickAppNotifications.qml which enables
or disables all notification types for the app. It gets automatically
switched when the user disables all notification types. Problem is, it
doesn't show up... it's behind the header!

This commit changes the page around so that the switch shows correctly.
It also makes the page scrollable and fixes a couple of papercuts I saw
from playing around with it. Now when the switch becomes disabled,
all of the notification-specific checkboxes don't display. When the
switch is enabled again, the checkboxes come back. Ahh, satisfying.

Closes ubports/ubuntu-touch#831

UniversalSuperBox committed:
Rejigger ClickAppNotifications page

Turns out there's a switch in ClickAppNotifications.qml which enables
or disables all notification types for the app. It gets automatically
switched when the user disables all notification types. Problem is, it
doesn't show up... it's behind the header!

This commit changes the page around so that the switch shows correctly.
It also makes the page scrollable and fixes a couple of papercuts I saw
from playing around with it. Now when the switch becomes disabled,
all of the notification-specific checkboxes don't display. When the
switch is enabled again, the checkboxes come back. Ahh, satisfying.

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -21,13 +21,16 @@ import Ubuntu.Components 1.3
-    title: entry ? entry.displayName : ""
+    header: PageHeader {
+        id: pageHeader
+        title: entry ? entry.displayName : ""
```
don't you need to add

    flickable: clickAppNotificationsFlickable

here?

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -21,13 +21,16 @@ import Ubuntu.Components 1.3
 import SystemSettings 1.0
 
-ItemPage {
+Page {
```
Is this change really needed, if you add the Flickable?

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -42,124 +45,143 @@ ItemPage {
                 }
             }
         }
+
```
remove this

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -42,124 +45,143 @@ ItemPage {
         }
+
     }
+
```
and this :-)

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -42,124 +45,143 @@ ItemPage {
-                                                                checked)
+    Flickable {
+        id: clickAppNotificationsFlickable
+        flickableDirection: Flickable.VerticalFlick
```
you need to set `contentHeight`, too

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -42,124 +45,143 @@ ItemPage {
+        flickableDirection: Flickable.VerticalFlick
+
+        anchors {
+            top: pageHeader.bottom
```
This should not be needed, the `Page` element will handle it itself. Just use `anchors.fill: parent`.

mardy reviewed (changes requested):
A few inline comments :-)

mardy commented:
See here for an example on how to use Page + Flickable + Column: https://gitlab.com/mardy/imaginario/blob/master/src/ubuntu-touch/qml/SettingsPage.qml

UniversalSuperBox commented:
Using ItemPage+Flickable+Column is *okay*, but it's very rare that this page goes past the bottom border of the screen. So, when you scroll down the header goes away. It doesn't make much sense for this page.

mardy commented:
Please elaborate a bit more :-) If it's rare for this page to go past the bottom of the screen, it means that it's also rare for this page to be scrolled and for the header to disappear, so I don't see a problem here.

UniversalSuperBox committed:
Suggestions for improvements to the page

Once the Flickable's contentHeight is set, the issue with being able to
swipe away the header magically goes away.

UniversalSuperBox commented:
It was my own misunderstanding of how to use these elements properly that caused the issue. I've addressed the requested changes and the page seems to work equally as well.

mardy commented on plugins/notifications/ClickAppNotifications.qml:
```diff
@@ -26,8 +26,8 @@ ItemPage {
 
     property var entry
     property int entryIndex
-
```
keep this line, it's a good practice to separe the public properties from the internal ones

mardy reviewed (changes requested):
Just a minor stylistic issue now :-)

UniversalSuperBox committed:
Separate public from internal properties

UniversalSuperBox commented:
There, that should fix that. I didn't know that style before, thanks!
